miden-lib/asm/miden/kernels/tx/note.masm

```diff
@@ -120,3 +130,157 @@ export.prepare_note
     exec.memory::get_consumed_note_script_root
     # => [NOTE_SCRIPT_ROOT]
 end
+
+# OUTPUT NOTE PROCEDURES
+# =================================================================================================
+
+#! Computes the assets hash of the output note located at the specified memory address. The hash
+#! is computed as a sequential hash of the assets contained in the note. If there is an odd number
+#! of assets, then for the final hashing permutation we pad the last word of the hasher rate with
+#! ZERO's.
+#!
+#! Stack: [note_data_ptr]
+#! Output: [ASSETS_HASH]
+#!
+#! - note_data_ptr is a pointer to the data section of the created note.
+#! - ASSETS_HASH is the hash of the assets of the output note located at note_data_ptr.
+proc.compute_output_note_assets_hash
+    # duplicate note pointer and fetch num_assets
+    dup dup exec.memory::get_created_note_num_assets
```

Comment:
Just for consistency, but should we always call created notes `output notes`? And the same for consumed notes?

Reply:
Yes, but in a future PR.